perun-core/src/main/java/cz/metacentrum/perun/core/impl/ConsentsManagerImpl.java

```diff
@@ -53,11 +63,289 @@
+		try {
+			List<Consent> consents = jdbc.query("select " + consentMappingSelectQuery + ", " + consentHubMappingSelectQuery +
+				" from consents left join consent_hubs on consents.consent_hub_id=consent_hubs.id " + " where consents.user_id=? ", CONSENT_MAPPER, id);
+			for (Consent consent : consents) {
+				if (consent != null) {
+					consent.getConsentHub().setFacilities(getFacilitiesForConsentHub(consent.getConsentHub()));
+					consent.setAttributes(getAttrDefsForConsent(sess, consent.getId()));
+				}
+			}
+			return consents;
+		} catch (EmptyResultDataAccessException ex) {
+			return new ArrayList<>();
+		} catch (RuntimeException err) {
+			throw new InternalErrorException(err);
+		}
+	}
+
+	@Override
+	public Consent getConsentById(PerunSession sess, int id) throws ConsentNotExistsException {
+		try {
+			Consent consent = jdbc.queryForObject("select " + consentMappingSelectQuery + ", " + consentHubMappingSelectQuery +
+				" from consents left join consent_hubs on consents.consent_hub_id=consent_hubs.id" + " where consents.id=? ", CONSENT_MAPPER, id);
+			if (consent != null) {
+				consent.getConsentHub().setFacilities(getFacilitiesForConsentHub(consent.getConsentHub()));
+				consent.setAttributes(getAttrDefsForConsent(sess, id));
+			}
+			return consent;
+		} catch (EmptyResultDataAccessException ex) {
+			throw new ConsentNotExistsException(ex);
+		} catch (RuntimeException err) {
+			throw new InternalErrorException(err);
+		}
+	}
+
+	@Override
+	public void checkConsentExists(PerunSession sess,  Consent consent) throws ConsentNotExistsException {
+		if (!consentExists(sess, consent)) throw new ConsentNotExistsException("Consent: " + consent);
+	}
+
+	@Override
+	public boolean consentExists(PerunSession sess, Consent consent) {
+		Utils.notNull(consent, "consent");
+		try {
+			int numberOfExistences = jdbc.queryForInt("select count(1) from consents where id=?", consent.getId());
+			if (numberOfExistences == 1) {
+				return true;
+			} else if (numberOfExistences > 1) {
+				throw new ConsistencyErrorException("Consent " + consent + " exists more than once.");
+			}
+			return false;
+		} catch (EmptyResultDataAccessException ex) {
+			return false;
+		} catch (RuntimeException ex) {
+			throw new InternalErrorException(ex);
+		}
+	}
+
+	private List<AttributeDefinition> getAttrDefsForConsent(PerunSession sess, int consentId) {
+		try {
+			return jdbc.query("select " + AttributesManagerImpl.attributeDefinitionMappingSelectQuery + " from attr_names join consent_attr_defs on attr_names.id = consent_attr_defs.attr_id" +
+				" where consent_attr_defs.consent_id=?", AttributesManagerImpl.ATTRIBUTE_DEFINITION_MAPPER, consentId);
+		} catch (EmptyResultDataAccessException ex) {
+			return new ArrayList<>();
+		} catch (RuntimeException err) {
+			throw new InternalErrorException(err);
+		}
+	}
+
+	private void createAttrDefsForConsent(PerunSession sess, Consent consent) throws ConsentNotExistsException {
```

Comment:
`ConsentNotExistsException` is not thrown here, I think. Please, remove it from the signature and all methods calling this one, too.